src/series/algorithm/waterfall.js

```diff
@@ -0,0 +1,100 @@
+import d3 from 'd3';
+import {noop} from '../../util/fn';
+
+export default function() {
+
+    var xValue = function(d) { return d.date; },
```

Comment:
The [long term vision](https://github.com/ScottLogic/d3fc/wiki/d3fc-Vision) is to split the d3fc core and financial concepts, for that reason I wouldn't use default values of `date` and `close` in the waterfall, I doubt many people render waterfall charts of financial data!